docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,471 @@
+# Machine Learning Training
+
+Machine Learning Training API allows you to construct, control, and train a machine learning model in Tizen devices.
+
+The main features of the Machine Learning Training API include:
+
+- Constructing deep neural network (DNN)
+   - You can construct a DNN model using a model description file or by writing code through Machine Learning Training API.
+- Training with your own data
+   - Machine Learning Training API also allows you to train the model with your own data as a File I/O or by defining a data generator.
+- Evaluating the model during train
+   - You can validate and test your model during the training process easily by defining the dataset.
+
+> [!NOTE]
+> Every example code does not handle all error use cases.
+> Error must be handled more extensively compared to the example code written in this page.
+
+## Prerequisites
+
+To enable your application to use Machine Learning Training API:
+
+1. To use the functions and data types of the Machine Learning Inference API, include the `<nntrainer.h>` header file in your application:
+
+   ```c
+   #include <nntrainer.h>
+   ```
+
+2. To use the Machine Learning Training API, include the following features in your `tizen-manifest.xml` file:
+
+   ```xml
+   <feature name="http://tizen.org/feature/machine_learning">true</feature>
+   <feature name="http://tizen.org/feature/machine_learning.training">true</feature>
+   ```
+
+In case of saving or loading model files from the outside of the application's own resources, the application has to request permission by adding the following privileges to the `tizen-manifest.xml` file:
+
+    ```xml
+    <privileges>
+      <!-- For accessing media storage -->
+      <privilege>http://tizen.org/privilege/mediastorage</privilege>
+      <!-- For accessing external storage -->
+      <privilege>http://tizen.org/privilege/externalstorage</privilege>
+    </privileges>
+    ```
+
+## Building blocks
+
+Following are the four major components of Machine Learning Training API:
+
+ - [Model](#model)
+ - [Layer](#layer)
+ - [Optimizer](#optimizer)
+ - [Dataset](#dataset)
+
+### Model
+
+
+Model is a wrapper component that has the topology of layers, optimizers, and datasets.
+The model performs training and saves the updated parameters that can later be used for inference.
+
+![model](./media/model.png)
+
+In this figure, `data` represent input data (or feature) and `label` is the actual value to be tested over prediction.
```

Comment:
Thanks for adding this sentence to describe `data / label`.

@surbhig12 Would you please also review this added sentence?

Reply:
I'd like to add one more comment to @surbhig12's suggestion. How about removing the parentheses in the sentence?

Model is a wrapper component that has the topology of layers, optimizers, and datasets.
The model performs training and saves the updated parameters that can later be used for inference. In the following figure, `data` represents input data or feature, and `label` is the actual value to be tested over prediction:

![model](./media/model.png)

Reply:
Updated as per @surbhig12, @mijin85cho 's suggestion